app/models/user.rb

```diff
@@ -1,2 +1,13 @@
 class User < ApplicationRecord
+  VALID_EMAIL_REGEX = Settings.valid_email_regex
+
+  before_save{self.email = email.downcase}
```

Comment:
```suggestion
  before_save{email.downcase!}
```
phân biệt downcase vs downcase! nhé

Reply:
downcase! sẽ thay đổi luôn giá trị ban đầu . còn downcase thì không

Reply:
ít thế e, tìm hiểu thêm

Reply:
downcase và downcase! đều dùng để chuyển xâu ký tự thành chữ thường
downcase! sẽ trả về nil nếu xâu ký tự không có gì thay đổi
VD: a = "hello"
a.downcase = "hello"
a.downcase! = nil
downcase! làm thay đổi giá trị xâu bạn đầu còn downcase thì không
VD:
a = "HEllo"
a.downcase = "hello"
puts a => HEllo
a = "HEllo"
a.downcase! = "hello"ello
puts a => hello